sdk/core/azure-core/test/ut/string_test.cpp

```diff
@@ -131,3 +131,52 @@ TEST(String, toUpper)
   EXPECT_NE(StringExtensions::ToUpper("a"), "aA");
   EXPECT_NE(StringExtensions::ToUpper("abc"), "abcd");
 }
+
+TEST(String, SplitEmpty)
+{
+  using Azure::Core::_internal::StringExtensions;
+  std::vector<std::string> result = StringExtensions::Split("", ',');
+  EXPECT_TRUE(result.empty());
+}
+
+TEST(String, Split3)
+{
+  using Azure::Core::_internal::StringExtensions;
+  std::vector<std::string> result = StringExtensions::Split("1,2,3", ',');
+  EXPECT_EQ(result[0], "1");
+  EXPECT_EQ(result[1], "2");
+  EXPECT_EQ(result[2], "3");
+  EXPECT_EQ(result.size(), size_t{3});
+}
+
+TEST(String, SplitBegin)
+{
+  using Azure::Core::_internal::StringExtensions;
+  std::vector<std::string> result = StringExtensions::Split(",1,2,3", ',');
+  EXPECT_EQ(result[0], "");
+  EXPECT_EQ(result[1], "1");
+  EXPECT_EQ(result[2], "2");
+  EXPECT_EQ(result[3], "3");
+  EXPECT_EQ(result.size(), size_t{4});
+}
+
+TEST(String, SplitEnd)
+{
+  using Azure::Core::_internal::StringExtensions;
+  std::vector<std::string> result = StringExtensions::Split("1,2,3,", ',');
+  EXPECT_EQ(result[0], "1");
+  EXPECT_EQ(result[1], "2");
+  EXPECT_EQ(result[2], "3");
+  EXPECT_EQ(result.size(), size_t{3});
```

Comment:
It might make sense for line 170 to be ASSERT_EQ() and move it above the EXPECT_EQ.

That way the test fails immediately on the ASSERT_EQ and we never get to the EXPECT_EQ (which will throw an exception if the ASSERT_EQ is not true).